config/hwloc.m4

```diff
@@ -873,7 +873,8 @@ EOF])
              AC_CHECK_HEADERS([X11/Xutil.h],
                 [AC_CHECK_HEADERS([X11/keysym.h],
                     [AC_DEFINE([HWLOC_HAVE_X11_KEYSYM], [1], [Define to 1 if X11 headers including Xutil.h and keysym.h are available.])])
-                     AC_SUBST([HWLOC_X11_LIBS], ["-lX11"])
+                     AC_SUBST([HWLOC_X11_CPPFLAGS], ["$(X_CFLAGS)"])
+                     AC_SUBST([HWLOC_X11_LIBS], ["$(X_PRE_LIBS) $(X_LIBS) $(X_EXTRA_LIBS) "])
```

Comment:
I'm not actually sure that the quotes are needed for HWLOC_X11_CPPFLAGS and HWLOC_X11_LIBS -- I kept them because they were in the original line 876.

But I can't check for sure because my system has X11 in default paths, so this code path isn't executed.  

@scottatchley Can you check to see what happens on your machine?